modules/webrequest-pipeline/sources/webrequest-context.es

```diff
@@ -128,6 +128,19 @@ export default class WebRequestContext {
   }
 
   isBackgroundRequest() {
-    return this.tabId === -1;
+    if (this.tabId !== -1) {
+      return false;
+    }
+    // On Firefox, requests from Service Workers have tabId !== -1.
+    // On Chromium, tabId === -1, so check additionally for the initiator.
+    // Requests from the service worker use a different protocol
+    // (e.g. https). If there is a more direct way to detect requests
+    // from the extension, this could be simplified.
+    if (isChromium && !this.initiator.startsWith('chrome-extensions://')) {
```

Comment:
what about Opera and Edge? maybe we should check for http/https instead?

Reply:
https://github.com/ghostery/common/blob/319cc4fd3c99db6733010b051663093c2959fa68/platforms/webextension/platform.es#L19

isChromium covers Chromium based browsers including Opera and Edge.

Reply:
http/https and ws/wss. But I tried to approach it from the other direction and identify requests from the extension. I think a better name for "isBackground" would be something in the line of "is it from the extension?".

Reply:
I've checked Opera and Edge and both use same `chrome-extensions` protocol

Reply:
Yes, I think the change here (not the URL filling) would be safe. On isolation, it will not make a difference though because the request will still be skipped because of the missing fields.